src/parse/resume.py

```diff
@@ -0,0 +1,72 @@
+import yaml
+import torch
+
+from train import TrainerSteps
+from .trainer import get_value, expand_curriculum, get_x_distribution, get_function_class, get_optimizer, get_loss_fn, get_model
+
+def parse_resume_training(content: str, latest_checkpoint_path: str, latest_step: int) -> TrainerSteps:
+    data = yaml.load(content, Loader=yaml.Loader)['train']
+
+    stages, step_counts = expand_curriculum(data)
+
+    resume_idx, steps, step = 0, [], 0
+    for step_count in step_counts:
+        step += step_count
+        steps.append(step)
+    steps_past = list(filter(lambda s: s <= latest_step, steps))
+    if not steps_past:
+        resume_idx = 0
+    else:
+        resume_idx = steps.index(steps_past[-1])
+    resume_stages, resume_step_counts = stages[resume_idx:], step_counts[resume_idx:]
```

Comment:
This seems to rely on checkpointing done in between training stages ... Since that's not true, I *feel* (not know) that it should fail for training runs with a checkpoint made before the end of a training stage.

Reply:
Some code now exists, needs to be verified for correctness